run_on_off_experiment.py

```diff
@@ -0,0 +1,249 @@
+    DEVICE = torch.device('cpu')
+
+
+def generate_plot(episodes, expert_ratios, scores, colour_len, log_freq, agent_type):
+    d = {"Episode": episodes, "ExpertRatio": expert_ratios}
+    experiments = pd.DataFrame(data=d)
+    experiments.insert(1, "Score", scores, allow_duplicates=False)
+    plot = sns.lineplot(data=experiments, x="Episode", y="Score", hue="ExpertRatio", legend="full",
+                        palette=sns.color_palette("hls", colour_len))
+    plot.set_xticks(range(0, max(episodes), log_freq))
+    plot.set_title(f"Scores of {agent_type}")
+
+    sns.lineplot(x=[ep for ep in range(0, max(episodes) + log_freq, log_freq)],
+                 y=[0 for _ in range(0, max(episodes) + log_freq, log_freq)],
+                 color="black", linestyle="--")
+
+    sns.lineplot(x=[ep for ep in range(0, max(episodes) + log_freq, log_freq)],
+                 y=[200 for _ in range(0, max(episodes) + log_freq, log_freq)],
+                 color="black", linestyle="--")
+
+    _path = os.path.join("experiments", "figures")
+    if not os.path.isdir(_path):
+        os.makedirs(_path, exist_ok=False)
+    figname = f"{agent_type}.png"
+    objname = f"{agent_type}.pkl"
+    f = open(os.path.join(_path, objname), 'wb')
+    pickle.dump(plot, f)
+    path = os.path.join(_path, figname)
+    plot.figure.savefig(path)
+
+
+def main(episodes_off, episodes_on, agent_type, num_trials):
+    # Log output
+    exp_name = "sac_on_off"
+    logdir = os.path.join("experiments", "logs", exp_name, agent_type)
+    os.makedirs(logdir, exist_ok=True)
+    log_freq = 100
+
+    env = gym.make('LunarLanderContinuous-v2')
+    n_states = env.observation_space.shape[0]  # shape returns a tuple
+    n_actions = env.action_space.shape[0]
+    agent = None
+    EXPERT_DATA_RATIOS = [0.0, 1.0]
+    exploration_eps = -1
+
+    total_episodes = []
+    total_expert_ratios = []
+    total_scores = []
+    for expert_data_ratio in EXPERT_DATA_RATIOS:
+        print(F"EXPERT RATIO: {expert_data_ratio}")
+        per_ratio_scores = []
+        per_ratio_logged_episodes = []
+        per_ratio_trials = []
+        per_ratio_logged_expert_ratios = []
+        for trial in range(num_trials):
+            print(F"CURR TRIAL: {trial}")
+            per_ratio_trials.append(trial)
+            ratio = (expert_data_ratio, 1.0 - expert_data_ratio)
+            if agent_type == "SACOffline":
+                agent = SAC(n_states=n_states, n_actions=n_actions, ratio=ratio)
+            elif agent_type == "BC":
+                agent = SAC(n_states=n_states, n_actions=n_actions, ratio=ratio, bc=True)
+            elif agent_type == "CQLSAC":
+                agent = SAC(n_states=n_states, n_actions=n_actions, ratio=ratio, cql=True, iS=True)
+
+            # Logging
+            exp_name = "_opt_ratio_" + str(expert_data_ratio) + "_trial_" + str(trial)
+            logdir2 = os.path.join(logdir, str(expert_data_ratio), str(trial))
+            os.makedirs(logdir2, exist_ok=True)
```

Comment:
same comment as my previous one about checking the existence of the directory first